gas-estimator/src/estimator.rs

```diff
@@ -0,0 +1,1222 @@
+                    match quote_price_exact_tokens_for_tokens(
+                        clients,
+                        dot_asset.clone(),
+                        ether_asset.clone(),
+                        delivery_fee_dot,
+                        true,
+                    )
+                    .await
+                    {
+                        Ok(delivery_fee_ether) => {
+                            (Some(delivery_fee_dot), Some(delivery_fee_ether))
+                        }
+                        Err(e) => {
+                            println!("Failed to convert destination delivery fee to Ether: {}", e);
+                            (Some(delivery_fee_dot), None)
+                        }
+                    }
+                }
+                Err(e) => {
+                    println!("Failed to calculate destination delivery fee: {}", e);
+                    (None, None)
+                }
+            }
+        } else {
+            (None, None)
+        };
+
+        match dry_run_xcm_on_destination(clients, forwarded_xcm).await {
+            Ok(dest_result) => (
+                Some(dest_result.success),
+                dest_result.error_message,
+                para_id,
+                dest_fee_dot,
+                dest_fee_ether,
+                dest_delivery_fee_dot,
+                dest_delivery_fee_ether,
+            ),
+            Err(e) => (
+                Some(false),
+                Some(format!("Destination dry run failed: {}", e)),
+                para_id,
+                dest_fee_dot,
+                dest_fee_ether,
+                dest_delivery_fee_dot,
+                dest_delivery_fee_ether,
+            ),
+        }
+    } else {
+        (None, None, None, None, None, None, None)
+    };
+
+    Ok(GasEstimation {
+        extrinsic_fee_in_dot,
+        extrinsic_fee_in_ether,
+        asset_hub: AssetHubInfo {
+            execution_fee_in_dot: asset_hub_execution_fee_in_dot,
+            execution_fee_in_ether: asset_hub_execution_fee_in_ether,
+            delivery_fee_in_dot,
+            delivery_fee_in_ether,
+            dry_run_success: ah_dry_run_result.success,
+            dry_run_error: ah_dry_run_result.error_message,
+        },
+        destination: DestinationInfo {
+            dry_run_success: destination_dry_run_success,
+            dry_run_error: destination_dry_run_error,
+            para_id: destination_para_id,
+            execution_fee_in_dot: destination_execution_fee_in_dot,
+            execution_fee_in_ether: destination_execution_fee_in_ether,
+            delivery_fee_in_dot: destination_delivery_fee_in_dot,
+            delivery_fee_in_ether: destination_delivery_fee_in_ether,
+        },
+    })
+}
+
+fn validate(value: u128, execution_fee: u128, relayer_fee: u128) -> Result<(), EstimatorError> {
+    if execution_fee.saturating_add(relayer_fee) > value {
+        return Err(EstimatorError::ValueTooLow);
+    }
+
+    Ok(())
+}
```

Comment:
Is this validation nessessary or correct? I'd assume the value in the payload [here](https://github.com/Snowfork/snowbridge/blob/c8c2e9399360ebf7f9b9d4765cb77976b15e3334/contracts/src/v2/Calls.sol#L111) is the original Ether amount minus (execution_fee + relayer_fee), so it's likely to be less than the total fee required.

Reply:
Good catch! Removed :)